caffe2/core/c10_operator.h

```diff
@@ -0,0 +1,96 @@
+#pragma once
+
+#include <vector>
+#include <ATen/core/dispatch/OpSchemaRegistration.h>
+#include <ATen/core/dispatch/KernelRegistration.h>
+#include <ATen/core/function_schema.h>
+
+namespace caffe2 {
+namespace detail {
+
+template<class Caffe2Operator> const c10::OperatorHandle& c10_op_handle_for_c2_op();
+template <class Caffe2Operator>
+void call_caffe2_op_from_c10(c10::Stack* stack, c10::KernelCache* cache) { // TODO Pass in correct cache type
+  // precondition: on the stack, there's an IValue for each input and an IValue for each output.
+  // The output ones could either be a preallocated tensor or ivalue::None.
+
+  const auto& schema = c10_op_handle_for_c2_op<Caffe2Operator>().schema();
+  const size_t num_outputs = schema.returns().size();
+  const size_t total_num_arguments = schema.arguments().size();
+  const size_t num_inputs = total_num_arguments - num_outputs;
+
+  c10::ArrayRef<c10::IValue> inputs = torch::jit::peekSlice(*stack, 0, num_inputs, total_num_arguments);
```

Comment:
So, like, you're about to drop them below; why not just pop the elements off the stack.